litmus-portal/backend/subscriber/agent/pkg/cluster/cluster.go

```diff
@@ -0,0 +1,111 @@
+package cluster
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	y "github.com/ghodss/yaml"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/serializer/yaml"
+	"k8s.io/client-go/discovery"
+	memory "k8s.io/client-go/discovery/cached"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/restmapper"
+	"log"
+)
+
+var decUnstructured = yaml.NewDecodingSerializer(unstructured.UnstructuredJSONScheme)
+
+func applyRequest(obj *unstructured.Unstructured, requestType interface{}, dr dynamic.ResourceInterface) (*unstructured.Unstructured, error) {
```

Comment:
Does this function handle any resource type[service,deployments,ns etc.] or specific resources only?

Reply:
Every resource type, its a dynamic client